bitcoind-rpc-test/src/test/scala/org/bitcoins/rpc/v17/PsbtRpcTest.scala

```diff
@@ -0,0 +1,223 @@
+package org.bitcoins.rpc.v17
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import akka.testkit.TestKit
+import org.bitcoins.core.config.NetworkParameters
+import org.bitcoins.core.currency.Bitcoins
+import org.bitcoins.core.number.UInt32
+import org.bitcoins.core.protocol.script.ScriptSignature
+import org.bitcoins.core.protocol.transaction.{
+  TransactionConstants,
+  TransactionInput,
+  TransactionOutPoint
+}
+import org.bitcoins.rpc.BitcoindRpcTestUtil
+import org.bitcoins.rpc.client.common.RpcOpts.AddNodeArgument
+import org.bitcoins.rpc.client.v17.BitcoindV17RpcClient
+import org.bitcoins.rpc.jsonmodels.{FinalizedPsbt, NonFinalizedPsbt}
+import org.scalatest.{AsyncFlatSpec, BeforeAndAfterAll}
+
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.concurrent.duration.DurationInt
+
+class PsbtRpcTest extends AsyncFlatSpec with BeforeAndAfterAll {
+  implicit val system: ActorSystem = ActorSystem(
+    "BitcoindV17RpcClientTest_ActorSystem")
+  implicit val m: ActorMaterializer = ActorMaterializer()
+  implicit val ec: ExecutionContext = m.executionContext
+  implicit val networkParam: NetworkParameters = BitcoindRpcTestUtil.network
+
+  val client = new BitcoindV17RpcClient(BitcoindRpcTestUtil.v17Instance())
+  val otherClient = new BitcoindV17RpcClient(BitcoindRpcTestUtil.v17Instance())
+  val thirdClient = new BitcoindV17RpcClient(BitcoindRpcTestUtil.v17Instance())
+
+  override protected def beforeAll(): Unit = {
```

Comment:
So here is a starting point for refactoring this

```scala
  val startF =
    BitcoindRpcTestUtil.startServers(Vector(client, otherClient, thirdClient))

  val pairsF = startF.map { _ =>
    List((client, otherClient), (client, thirdClient))
  }

  val addNodesF: Future[List[Unit]] = {
    pairsF.flatMap { pairs =>
      val addedF = pairs.map {
        case (first, second) =>
          first.addNode(second.getDaemon.uri, AddNodeArgument.Add)
      }
      Future.sequence(addedF)
    }
  }

  val connectedF: Future[List[(BitcoindV17RpcClient, BitcoindV17RpcClient)]] = {
    pairsF.flatMap { pairs =>
      val connectedPairsF = {
        pairs.map {
          case (first, second) =>
            val connF =
              BitcoindRpcTestUtil.awaitConnectionF(first,
                                                   second,
                                                   duration = 1.second)
            connF.map(_ => (first, second))
        }
      }
      Future.sequence(connectedPairsF)
    }
  }
```

and i think this should be moved outside of the `beforeAll`. I also took the liberty of `BitcoindRpcTestUtil.awaitConnection` to these two methods now which i use above:

```scala
  def awaitConnection(
      from: BitcoindRpcClient,
      to: BitcoindRpcClient,
      duration: FiniteDuration = 100.milliseconds,
      maxTries: Int = 50)(implicit system: ActorSystem): Unit = {
    implicit val ec: ExecutionContextExecutor = system.dispatcher
    val connF = awaitConnectionF(from, to, duration, maxTries)
    Await.result(connF, 1.hour)
  }

  def awaitConnectionF(
      from: BitcoindRpcClient,
      to: BitcoindRpcClient,
      duration: FiniteDuration = 100.milliseconds,
      maxTries: Int = 50)(implicit system: ActorSystem): Future[Unit] = {
    import system.dispatcher
    def isConnected(): Future[Boolean] = {
      from
        .getAddedNodeInfo(to.getDaemon.uri)
        .map { info =>
          info.nonEmpty && info.head.connected.contains(true)
        }
    }

    AsyncUtil.retryUntilSatisfiedF(conditionF = isConnected,
                                   duration = duration,
                                   maxTries = maxTries)
  }
```